packages/medusa/src/loaders/module.ts

```diff
@@ -42,15 +47,50 @@ const registerModule = async (
     }
   }
 
+  if (
+    resolution.moduleDeclaration?.scope === MODULE_SCOPE.INTERNAL &&
+    resolution.moduleDeclaration?.resources === MODULE_RESOURCE_TYPE.SHARED
+  ) {
+    const moduleModels = loadedModule?.models || null
+    if (moduleModels) {
+      moduleModels.map((val: ClassConstructor<unknown> | EntitySchema) => {
+        const name = /function (\w+)/.exec(val.toString())?.[1]
```

Comment:
todo: on a class you can get Class.name or on an instance Class.constructor.name

Reply:
Thanks. constructor.name was returning Function and I quickly added this to test.
.name does the trick, but I will remove all that to do not register models to the core container